logger-service/logger-service.js

```diff
@@ -0,0 +1,82 @@
+const fsExtra = require('fs-extra');
+const outputCache = new Map();
+
+const AppenderTypes = {
+	NONE: 'none',
+	NORMAL: 'normal',
+	STRUCTURED: 'structured'
+};
+
+const outputParameterStoragePath = `${process.cwd()}/.outputParameter`;
+
+class Logger {
+
+	constructor() {
+	}
+
+	log() {
+		const appenderType = this.getOutputParameterValue();
+		const args = Array.prototype.slice.call(arguments);
+
+		if (appenderType === AppenderTypes.NONE) {
+			return;
+		}
+
+		if (appenderType === AppenderTypes.STRUCTURED) {
+			this.logToFile(args);
+			return;
+		}
+
+		this.logToConsole(args);
+	}
+
+	logToConsole(args) {
+		console.log.apply(console, args);
+	}
+
+	logToFile(args) {
+		//ToDO: Log to file implementation
+	}
+
+	storeOutputParameter(value) {
+		fsExtra.writeFileSync(outputParameterStoragePath, value);
+		this.updateOutputCache(value);
+	}
+
+	getOutputParameterValue() {
+		let fileContent;
+
+		if (outputCache.has(outputParameterStoragePath)) {
+			fileContent = outputCache.get(outputParameterStoragePath);
+			return fileContent;
+		}
+
+		if (!fsExtra.existsSync(outputParameterStoragePath)) {
+			return AppenderTypes.NONE;
+		}
+
+		fileContent = fsExtra.readFileSync(outputParameterStoragePath).toString('utf-8');
+		this.updateOutputCache(fileContent);
+
+		return fileContent;
+	}
+
+	updateOutputCache(value) {
+		outputCache.set(outputParameterStoragePath, value);
```

Comment:
As you are storing the output file in the directory of the user project, I do not thing we will ever need more than one value. I think just storing the value in a string variable might be easier and cleaner